src/components/ReanimatedDrawerLayout.tsx

```diff
@@ -0,0 +1,716 @@
+                [-drawerWidth - 1, -drawerWidth, 0],
+                [drawerWidth, drawerWidth, 0]
+              );
+
+          const startedInsideTranslation =
+            sideCorrection *
+            (event.translationX +
+              (drawerOpened ? drawerWidth * -gestureOrientation : 0));
+
+          dragX.value =
+            sideCorrection *
+            Math.max(
+              drawerOpened ? startedOutsideTranslation : 0,
+              startedInsideTranslation
+            );
+
+          drawerTranslation.value =
+            drawerType === DrawerType.FRONT
+              ? sideCorrectedDragX.value +
+                interpolate(
+                  -1 * sideCorrectedDragX.value,
+                  [drawerWidth - 1, drawerWidth, drawerWidth + 1],
+                  [0, 0, 1]
+                )
+              : sideCorrectedDragX.value;
+        })
+        .onEnd(handleRelease);
+    }, [
+      dragX,
+      drawerLockMode,
+      drawerTranslation,
+      drawerType,
+      drawerWidth,
+      emitStateChanged,
+      gestureOrientation,
+      handleRelease,
+      edgeHitSlop,
+      fillHitSlop,
+      minSwipeDistance,
+      props.hideStatusBar,
+      props.keyboardDismissMode,
+      sideCorrectedDragX.value,
+      overlayDismissGesture,
+      drawerOpened,
+      isFromLeft,
+      containerWidth,
+      sideCorrection,
+      drawerState,
+      activeCursor,
+      enableTrackpadTwoFingerGesture,
+      mouseButton,
+      statusBarAnimation,
+    ]);
+
+    // When using RTL, row and row-reverse flex directions are flipped.
+    const reverseContentDirection = I18nManager.isRTL
+      ? isFromLeft
+      : !isFromLeft;
+
+    const dynamicDrawerStyles = {
+      backgroundColor: drawerBackgroundColor,
+      width: drawerWidth,
+    };
+
+    const containerStyles = useAnimatedStyle(() => {
+      if (drawerType === DrawerType.FRONT) {
+        return {};
+      }
+
+      return {
+        transform: [
+          {
+            translateX: interpolate(
+              openValue.value,
+              [0, 1],
+              [0, drawerWidth * sideCorrection],
+              Extrapolation.CLAMP
+            ),
+          },
+        ],
+      };
+    });
+
+    const drawerAnimatedStyle = useAnimatedStyle(() => {
+      const closedDrawerOffset = drawerWidth * -sideCorrection;
+      const isBack = drawerType === DrawerType.BACK;
+      const isIdle = drawerState === DrawerState.IDLE;
+
+      let translateX = 0;
+
+      if (!isBack && isIdle) {
+        translateX = drawerOpened ? 0 : closedDrawerOffset;
+      } else if (!isBack && !isIdle) {
+        translateX = interpolate(
+          openValue.value,
+          [0, 1],
+          [closedDrawerOffset, 0],
+          Extrapolation.CLAMP
+        );
+      }
+
+      return {
+        transform: [
+          {
+            translateX,
+          },
+        ],
+        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
+      };
```

Comment:
Actually in that case we could simply check if `isBack` is `true` and return right away.

```suggestion
      if (isBack) {
        return {
          transform: [{ translateX: 0 }],
          flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
        };
      }

      let translateX = 0;

      if (isIdle) {
        translateX = drawerOpened ? 0 : closedDrawerOffset;
      } else {
        translateX = interpolate(
          openValue.value,
          [0, 1],
          [closedDrawerOffset, 0],
          Extrapolation.CLAMP
        );
      }

      return {
        transform: [{ translateX }],
        flexDirection: reverseContentDirection ? 'row-reverse' : 'row',
      };
```

WDYT?

Reply:
Sure, looks great, changed in [ea31eb3](https://github.com/software-mansion/react-native-gesture-handler/pull/3146/commits/ea31eb34c2ba31113b0ab71ec848537b1204d3d6).